CONTRIBUTING.md

```diff
@@ -180,12 +179,6 @@ python3 -m pytest tests/functional/sources
 
 > See [pytest usage docs](https://docs.pytest.org/en/6.2.x/usage.html) for an overview of useful command-line options.
 
-### Unit, Integration, Functional?
-
-Here are some general rules for adding tests:
-* unit tests (`tests/unit`) don’t need to access a database; "pure Python" tests should be written as unit tests
-* functional tests (`tests/functional`) cover anything that interacts with a database, namely adapter
-
```

Comment:
We should keep this section as it's specific to `dbt-core` and the 2 types of tests we use.